src/ServiceControl.Transports/TransportCustomization.cs

```diff
@@ -33,6 +35,10 @@ public abstract class TransportCustomization<TTransport> : ITransportCustomizati
 
         protected abstract void CustomizeTransportForMonitoringEndpoint(EndpointConfiguration endpointConfiguration, TTransport transportDefinition, TransportSettings transportSettings);
 
+        public virtual void AddTransport(IServiceCollection services)
```

Comment:
@danielmarbach, if we are making this seam aware of asp.net/di, it makes me wonder if the `IProvideQueueLength` could be turned into a `BackgroundService` and registered accordingly. Thoughts?
It would be one less abstraction that we need to take care.

Reply:
Yes that would be my idea as well